src/compiler/ast_native.cc

```diff
@@ -642,6 +642,11 @@ class ASTNativeCompilerImpl {
           "split_index"_a = node->split_index,
           "right_categories_flag"_a = right_categories_flag);
       }
+
+      oss << fmt::format(
+          "((data[{split_index}].fvalue >= 0) && !isinf(data[{split_index}].fvalue) &&"
+          "(fabsf(data[{split_index}].fvalue) <= (float)(1U << FLT_MANT_DIG)) && (",
+          "split_index"_a = node->split_index);
```

Comment:
and what do we do with LGBM not conforming? this does look like the ideal behavior, should we just merge this and wait until they start throwing exceptions on invalid categories?

Reply:
@levsnv I think we can persuade them to clarify the behavior for INFs and large floats, since the concern for backward compatibility is lesser.

Reply:
sounds perfect!